pkg/sql/sqlstats/insights/integration/integration_test: TestInsightsIntegration failed

pkg/sql/sqlstats/insights/integration/integration_test.TestInsightsIntegration [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/12310446?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/12310446?buildTab=artifacts#/) on master @ [3c05cb0a81d9916c010bc71dbc91c1c686a9febd](https://github.com/cockroachdb/cockroach/commits/3c05cb0a81d9916c010bc71dbc91c1c686a9febd):


```
=== RUN   TestInsightsIntegration
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/8db7e2fbdb77cb910d7e66552998132e/logTestInsightsIntegration4023723680
    test_log_scope.go:81: use -show-logs to present logs inline
    test_server_shim.go:124: automatically injected virtual cluster under test; see comment at top of test_server_shim.go for details.
    insights_test.go:89: condition failed to evaluate within 1s: expected 1, but was 2, queryText:SELECT pg_sleep($1);SELECT count(*), COALESCE(string_agg(query, '_'), '_') FROM crdb_internal.cluster_execution_insights WHERE app_name = $1
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/8db7e2fbdb77cb910d7e66552998132e/logTestInsightsIntegration4023723680
--- FAIL: TestInsightsIntegration (5.35s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/cluster-observability
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestInsightsIntegration.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-32655

--- a/pkg/sql/sqlstats/insights/integration/insights_test.go
+++ b/pkg/sql/sqlstats/insights/integration/insights_test.go
@@ -55,6 +55,7 @@ func TestInsightsIntegration(t *testing.T) {
 	defer log.Scope(t).Close(t)
 
 	const appName = "TestInsightsIntegration"
+	const appNameToIgnore = "TestInsightsIntegrationIgnore"
 
 	// Start the cluster. (One node is sufficient; the outliers system is currently in-memory only.)
 	ctx := context.Background()
@@ -68,7 +69,7 @@ func TestInsightsIntegration(t *testing.T) {
 	latencyThreshold := 250 * time.Millisecond
 	insights.LatencyThreshold.Override(ctx, &settings.SV, latencyThreshold)
 
-	_, err := conn.ExecContext(ctx, "SET SESSION application_name=$1", appName)
+	_, err := conn.ExecContext(ctx, "SET SESSION application_name=$1", appNameToIgnore)
 	require.NoError(t, err)
 
 	// See no recorded insights.
@@ -81,8 +82,16 @@ func TestInsightsIntegration(t *testing.T) {
 	require.Equal(t, 0, count, "expect:0, actual:%d, queries:%s", count, queryText)
 
 	queryDelayInSeconds := latencyThreshold.Seconds()
+	_, err = conn.ExecContext(ctx, "SET SESSION application_name=$1", appName)
+	require.NoError(t, err)
+
 	// Execute a "long-running" statement, running longer than our latencyThreshold.
+	// Use a specific app name just for this query, and ignore the other statements, since
+	// the select used to check the values can end up or not as part of the Insights view.
 	_, err = conn.ExecContext(ctx, "SELECT pg_sleep($1)", queryDelayInSeconds)
+	require.NoError(t, err)
+
+	_, err = conn.ExecContext(ctx, "SET SESSION application_name=$1", appNameToIgnore)
 	require.NoError(t, err)
 
 	// Eventually see one recorded insight.

